### ISSUE TITLE
java_test fails with un-insightful error message in Bazel 7

### ISSUE THREAD

**Issue by joshua-k-harris** [bazelbuild/bazel#20378]
### Description of the bug:

The `java_test` rule fails with an un-insightful error message, when the `test_class` can't be
determined in the latest Bazel 7 RC for a `java_test` target at the root of a repository. 

The error message displayed is: `path must not be absolute: '/some_test'` but should be something more informative like`cannot determine test class`.

### Which category does this issue belong to?

Java Rules

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. Ensure the latest Bazel 7 RC is being used. Ensure the `.bazelverison` file has the following content.
```starlark
last_rc
```

2. In A `BUILD.bazel` file at the **_root_** of the repository, add the following `java_test` target.
```starlark
java_test(
  name = "some_test",
  srcs = ["SomeTest.java"],
)
```

3. Run the `:some_test` `java_test` target as a test target.
```starlark
bazel test :some_test
```

The stacktrace below is what's outputted by Bazel.
```
Traceback (most recent call last):
        File "/virtual_builtins_bzl/bazel/java/bazel_java_binary.bzl", line 28, column 35, in _bazel_java_test_impl
        File "/virtual_builtins_bzl/bazel/java/bazel_java_binary.bzl", line 70, column 46, in _bazel_base_binary_impl
        File "/virtual_builtins_bzl/common/java/java_helper.bzl", line 96, column 27, in _primary_class
        File "/virtual_builtins_bzl/common/java/java_helper.bzl", line 105, column 35, in _full_classname
        File "/virtual_builtins_bzl/common/java/java_helper.bzl", line 110, column 13, in _java_segments
Error in fail: path must not be absolute: '/some_test'

```

### Which operating system are you running Bazel on?

macOS (arm64)

### What is the output of `bazel info release`?

release 7.0.0rc4

### Is this a regression? If yes, please try to identify the Bazel commit where the bug was introduced.

Yes this is a regression. This does not occur in Bazel 6.4.0.

**Comment by illicitonion** [comment]
@comius this looks like an accidental consequence of starlarkification of java rules?

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.0.0 RC5](https://releases.bazel.build/7.0.0/rc5/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
